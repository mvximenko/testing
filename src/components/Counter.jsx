import { useState } from 'react';

const Counter = () => {
  const [count, setCount] = useState(0);

  return (
    <div>
      <button onClick={() => setCount(count + 1)}>+</button>
      <button onClick={() => setCount(count - 1)}>-</button>
      <p>Count is {count}</p>
    </div>
  );
};

export default Counter;
