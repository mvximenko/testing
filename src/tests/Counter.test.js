import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';
import Counter from '../components/Counter';

describe('Counter', () => {
  it('increments counter', async () => {
    const user = userEvent.setup();

    render(<Counter />);

    // paragraph is not supported yet: https://github.com/testing-library/dom-testing-library/issues/1234
    // const paragraph = screen.getByRole('paragraph');
    const paragraph = screen.getByText(/count/i);
    const button = screen.getByRole('button', { name: '+' });

    await user.click(button);

    expect(paragraph).toHaveTextContent('Count is 1');
  });

  it('decrements counter', async () => {
    const user = userEvent.setup();

    render(<Counter />);

    const paragraph = screen.getByText(/count/i);
    const button = screen.getByRole('button', { name: '-' });

    await user.click(button);

    expect(paragraph).toHaveTextContent('Count is -1');
  });
});

describe('Counter Snapshot', () => {
  it('matches DOM Snapshot', () => {
    const tree = renderer.create(<Counter />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
