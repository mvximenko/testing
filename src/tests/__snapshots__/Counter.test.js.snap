// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Snapshot matches DOM Snapshot 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    +
  </button>
  <button
    onClick={[Function]}
  >
    -
  </button>
  <p>
    Count is 
    0
  </p>
</div>
`;
